File: frontend/src/api/chat/dto/GetChatRooms.ts
export interface GetChatRoomsResponse {
  chatRooms: ChatRoom[]
}

interface ChatRoom {
  id: number
  chatParticipantNickname: string
  rentalItem: RentalItem
  messageStatus: MessageStatus
}

interface RentalItem {
  title: string
  thumbnailImageUrl: string
}

interface MessageStatus {
  latestMessage: string
  latestMessageTime: Date
  unreadCount: number
}

File: backend/api/src/main/kotlin/io/github/daegwonkim/backend/dto/chatroom/GetChatRooms.kt
package io.github.daegwonkim.backend.dto.chatroom

import java.time.Instant

data class GetChatRoomsResponse(
    val chatRooms: List<ChatRoom>
) {
    data class ChatRoom(
        val id: Long,
        val chatParticipantNickname: String,
        val rentalItem: RentalItem,
        val messageStatus: MessageStatus
    ) {
        data class RentalItem(
            val title: String,
            val thumbnailImageUrl: String,
        )

        data class MessageStatus(
            val latestMessage: String,
            val latestMessageTime: Instant,
            val unreadCount: Int
        )
    }
}